From 3b40f1c1434f8705cb32e74729f520f610cb95cf Mon Sep 17 00:00:00 2001
From: Aryan Sinha <techyminati@outlook.com>
Date: Mon, 13 Apr 2026 09:27:04 +0200
Subject: [PATCH 1/3] SystemUI: Fix duplicate provider method for MicroG tile

Rename provideVpnTileConfig() to provideMicrogTileConfig() in
LineageModule to avoid duplicate binding method names in Dagger.

This fixes build failure:
"Cannot have more than one binding method with the same name
in a single module"
---
 .../SystemUI/src/com/android/systemui/lineage/LineageModule.kt  | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/lineage/LineageModule.kt b/packages/SystemUI/src/com/android/systemui/lineage/LineageModule.kt
index 5610f2b21eb5..359f8ce2345e 100644
--- a/packages/SystemUI/src/com/android/systemui/lineage/LineageModule.kt
+++ b/packages/SystemUI/src/com/android/systemui/lineage/LineageModule.kt
@@ -275,7 +275,7 @@ interface LineageModule {
        @Provides
        @IntoMap
        @StringKey(MICROG_TILE_SPEC)
-       fun provideVpnTileConfig(uiEventLogger: QsEventLogger): QSTileConfig =
+       fun provideMicrogTileConfig(uiEventLogger: QsEventLogger): QSTileConfig =
            QSTileConfig(
                tileSpec = TileSpec.create(MICROG_TILE_SPEC),
                uiConfig =
-- 
2.34.1

